components/service/glean/src/main/java/mozilla/components/service/glean/Glean.kt

```diff
@@ -504,25 +504,25 @@ open class GleanInternalAPI internal constructor () {
     /**
      * Send a list of pings.
      *
-     * The ping content is assembled as soon as possible, but upload is not
-     * guaranteed to happen immediately, as that depends on the upload
-     * policies.
+     * The ping content is assembled synchronously, but serialization to disk
+     * happens in a deferred coroutine, and upload happens later as that depends
+     * on the upload policies.
      *
      * If the ping currently contains no content, it will not be sent.
      *
      * @param pings List of pings to send.
      * @return The async Job performing the work of assembling the ping
      */
     @Suppress("EXPERIMENTAL_API_USAGE")
-    internal fun sendPings(pings: List<PingType>) = Dispatchers.API.launch {
+    internal fun sendPings(pings: List<PingType>) {
```

Comment:
This turned out to be structured _exactly_as on [desktop](https://searchfox.org/mozilla-central/rev/f372e8a46ef7659ef61be9938ec2a3ea34d343c6/toolkit/components/telemetry/app/TelemetryController.jsm#502). I think this is a good call.